src/brod_consumer.erl

```diff
@@ -48,238 +42,209 @@
         , handle_info/2
         , terminate/2
         , code_change/3
-        , format_status/2
         ]).
 
 -include("brod.hrl").
 -include("brod_int.hrl").
 
--record(state, { hosts         :: [{string(), integer()}]
-               , socket        :: #socket{}
+-type consumer_option() :: offset
```

Comment:
The only option which I know exactly will be used is 'offset' – when users will use their own storage for offset, in the 'init_consumer' callback they will query offset for the given topic/partition pair.
Other options can be used to override global options for specific topic/partitions.